### ISSUE TITLE
White 1 pixel right margin always displayed in generated pdf's

### ISSUE THREAD

**Issue by cody-vibe** [wkhtmltopdf/wkhtmltopdf#1982]
To reproduce type the following html

```
<!DOCTYPE html>
<html style="width: 100%; height: 100%; background: blue;">
<head>
</head>
<body>
</body>
</html>
```

and execute the following command:

```
wkhtmltopdf --margin-top 0 --margin-bottom 0 --margin-left 0 --margin-right 0 test.html test.pdf
```

The resulted PDF has a 1 pixel white border in the right side. 

I am using: `wkhtmltopdf 0.12.1 (with patched qt)`

**Comment by jerefrer** [comment]
Adding `--page-height 297mm --page-width 212mm` just worked. Thanks a lot :wink:

**Comment by MarcReniu** [comment]
But by doing this you are modifing the standard A4 page size, that has to be 297mm by 210mm. This is not the "proper" solution.

**Comment by heraldsuva** [comment]
I have every possible solutions/suggestion from every forums, one of that is setting the page size to 297mmx212mm. Still the tiny white border is present in each corner.

**Comment by MarcReniu** [comment]
@heraldsuva for one page print could be an affordable solution. When you have a bunch, you are adding an space error for each one and . by the 10th page, you will see a huge white border.

**Comment by 1pavanb** [comment]
Issue still persists

wkhtmltopdf 0.12.6 (with patched qt)
for A4 size
```
echo "<html style='padding=0;margin=0'><body style='background-color:black;padding=0;margin=0'></html>" | wkhtmltopdf -T 0 -B 0 -L 0 -R 0 --disable-smart-shrinking - bad.pdf
```

with page width extended like the people above suggested
```
echo "<html style='padding=0;margin=0'><body style='background-color:black;padding=0;margin=0'></html>" | wkhtmltopdf -T 0 -B 0 -L 0 -R 0 --disable-smart-shrinking  --page-height 297mm --page-width 212mm - anotherbad.pdf
```

**Comment by 1pavanb** [comment]
Solved it by increasing the DPI

I'm working with an A4 size in portrait mode. Had white space to the right.


I noticed that as the dpi is increased, the white space got thinner.
at 300 dpi the white space is not visible in chrome pdf view at (max) zoomed at 500%

In Adobe reader it's still visible. It got better at 600 DPI and at 1200 DPI it's become invisible even at 6500% zoom.

There's no disadvantage to this so far as I observed, all dpi generate the same file size and run at the same speed (tested on 1 page).

effectively my settings are as follows:

    echo "<html style='padding=0;margin=0'><body style='background-color:black;padding=0;margin=0'></html>" | wkhtmltopdf -T 0 -B 0 -L 0 -R 0 --disable-smart-shrinking --orientation portrait --page-size A4 --dpi 1200 - happy.pdf
